pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -501,10 +501,18 @@ def _get_monitor_value(self, key: str) -> Any:
         return self.trainer.callback_metrics.get(key)
 
     def _should_check_val_epoch(self):
-        return (
-            self.trainer.enable_validation
-            and (self.trainer.current_epoch + 1) % self.trainer.check_val_every_n_epoch == 0
-        )
+        if not self.trainer.enable_validation:
+            return False
+
+        # first we check if `check_val_every_n_epoch is `None`, which means
+        # that we run a validation loop after n global steps (taken from the
+        # Trainer argument `val_check_interval`)
+        if self.trainer.check_val_every_n_epoch is None:
+            return (self.trainer.global_step + 1) % self.trainer.val_check_batch == 0
```

Comment:
I believe there is no case where validation runs before any optimizer steps, so thhis should be

```suggestion
            return self.trainer.global_step % self.trainer.val_check_batch == 0
```